packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -225,6 +225,45 @@ export interface CanaryProps {
    */
   readonly test: Test;
 
+  /**
+   * Key-value pairs that the Synthetics caches and makes available for your canary
+   * scripts. Use environment variables to apply configuration changes, such
+   * as test and production environment configurations, without changing your
+   * Canary script source code.
+   *
+   * @default - No environment variables.
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * The function execution time (in seconds) after which Lambda terminates
+   * the function. Because the execution time affects cost, set this value
+   * based on the function's expected execution time.
+   *
+   * @default cdk.Duration.seconds(900)
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * The amount of memory, in MB, that is allocated to your Lambda function.
+   * Lambda uses this value to proportionally allocate the amount of CPU
+   * power. For more information, see Resource Model in the AWS Lambda
+   * Developer Guide.
+   *
+   * @default 128
+   */
+  readonly memorySize?: number;
```

Comment:
This value must be a multiple of 64. The range is 960 to 3008. 

See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-memoryinmb

Reply:
I added it to the doc but not sure if I should check the `% 64 === 0` and throw. I haven't seen such things being done anywhere else in cdk source code. Let me know if I should add the check and throw.

Reply:
this should use Size from @aws-cdk/core and also please add checks for the multiple of 64 and the min and max values